ui/app/routes/vault/cluster/secrets/backend/credentials.js

```diff
@@ -64,15 +67,23 @@ export default Route.extend({
     }
   },
 
+  getTOTPCode(backend, keyName) {
+    return this.store.adapterFor('totp').generateCode(backend, keyName);
+  },
+
   async model(params) {
     const role = params.secret;
     const { id: backendPath, type: backendType } = this.modelFor('vault.cluster.secrets.backend');
     const roleType = params.roleType;
-    let dbCred, awsRole;
+    let dbCred, awsRole, totpCode;
     if (backendType === 'database') {
       dbCred = await this.getDatabaseCredential(backendPath, role, roleType);
     } else if (backendType === 'aws') {
       awsRole = await this.getAwsRole(backendPath, role);
+    } else if (backendType === 'totp') {
+      totpCode = await this.getTOTPCode(backendPath, role);
+      const totpKey = await this.store.queryRecord('totp', { id: role, backend: backendPath });
```

Comment:
Just flagging that doing it this way means a user with only generate permissions in their policy will get a failure on this page because it will also attempt to read the key. If we wanted to make the period optional you could do something similar to the `getAwsRole` method to fail silently on the non-essential data